katka/signals.py

```diff
@@ -0,0 +1,33 @@
+from django.conf import settings
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from .constants import PIPELINE_STATUS_INITIALIZING, STEP_FINAL_STATUSES
+from .fields import username_on_model
+from .models import SCMPipelineRun, SCMStepRun
+
+
+@receiver(post_save, sender=SCMStepRun)
+def update_pipeline_steps_completed(sender, **kwargs):
+    pipeline = kwargs['instance'].scm_pipeline_run
+    pipeline_steps = SCMStepRun.objects.filter(scm_pipeline_run=pipeline)
+
+    before_steps_total = pipeline.steps_total
+    before_steps_completed = pipeline.steps_completed
+
+    pipeline.steps_total = pipeline_steps.count()
+    pipeline.steps_completed = pipeline_steps.filter(status__in=STEP_FINAL_STATUSES).count()
+
+    if pipeline.steps_completed != before_steps_completed or pipeline.steps_total != before_steps_total:
+        with username_on_model(SCMPipelineRun, kwargs['instance'].modified_username):
+            pipeline.save()
+
+
+@receiver(post_save, sender=SCMPipelineRun)
+def send_pipeline_change_notification(sender, **kwargs):
+    pipeline = kwargs['instance']
+    if pipeline.status == PIPELINE_STATUS_INITIALIZING:
```

Comment:
Is pipeline status updating for a different PR?

Reply:
No, it's part of this PR. But I'll add a comment to clarify why we do not send updates for initializing.

Reply:
Okay, then who puts the pipeline status in progress? Is this done by the pipeline runner?

Reply:
yes, it is also the one initializing it, so it should put it in progress as well